klsh/hamming_ann.py

```diff
@@ -156,12 +158,14 @@ def query(self, X, k, return_dist=False):
             dist = np.zeros_like(nbrs)
 
         M, N = self._P_compact_sorted.shape
-
+        
+        X_compact=packbits_axis(X_compact[:,self._P_indices])
+        
         # TODO: MAKE THIS MORE EFFICIENT
         for i, val in enumerate(X_compact):
             # find ordered index within each random permutation
             P_indices = np.array([np.searchsorted(self._P_compact_sorted[j],
-                                                  val) for j in range(M)])
+                                                  val[j]) for j in range(M)])
```

Comment:
In the earlier code` val `variable contains query bits without permutation and they were compared with permuted training data bits. In this modified code `val[j]` contains query bits for the `j`th permutation and they are compared with the` j`th permutation of the training data and indices of the nearest neighbor is stored in `P_indices`. Variable val contains all the `M `permutation of query bits.